mitodl/ocw-hugo-themes
# Updated promo carousel to handle featured promos

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes #1052 

#### What's this PR do?
This PR accomplishes the following
- Updates the promos-carousal to handle featured promos
- Adds check to ensure carousel does not render in case of absence of promos
- Adds a partial which gets pages based on a slice of UIDs

#### How should this be manually tested?
- Add a homepage.yml in data folder of ocw-www content in the following format
```
  featured_promos:
    content:
    - 9cfbaac6-a953-402a-92c3-5ac632d2fc99
    - 266b7de6-5f8d-44c8-ade8-a2ffa1735d64
    website: test-site
```
- Verify that the promos on home screen appear in the same order as described in homepage.yml 


#### Any background context you want to provide?

This PR is part of the solution recommended by @ChristopherChudzicki on his latest [comment](https://github.com/mitodl/ocw-hugo-projects/issues/183#issuecomment-1382371103) on [discussion](https://github.com/mitodl/ocw-hugo-projects/issues/183) ). This PR just updated ocw-www theme to filter and display promos based on featured promos in the specified order.


#### Screenshots (if appropriate)
(Optional)
<img width="1440" alt="Screenshot 2023-01-24 at 5 53 15 PM" src="https://user-images.githubusercontent.com/87968618/214307763-2932f65e-e3ea-41fa-9567-8af49950d96d.png">
<img width="1440" alt="Screenshot 2023-01-24 at 5 53 15 PM" src="https://user-images.githubusercontent.com/87968618/214307938-60712a4f-c3e3-4fd0-b802-012ac31552e8.png">
